### PR TITLE
add defaults to cluster_bootstrap.py

### PR DESCRIPTION
**Pull request by LucasWang750**
### Description
add defaults to cluster_bootstrap.py

### Issues Resolved
Closes [#65]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by LucasWang750** [comment]
Since dictionaries are mutable. There might've been a risk of default dict parameter being mutated so I used None instead.

File: demo/docbot/cluster_bootstrap.py [hunk 1]
```diff
@@ -17,32 +17,32 @@
 
 
 
-def initialize_cluster_settings(client: OpenSearch) -> None:
+def initialize_cluster_settings(client: OpenSearch, cluster_setting=None) -> None:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:20]: Here can we remove the template parameters? We really just need to pull out the parts that may be variable. For this one I think they can all stay.

File: demo/docbot/cluster_bootstrap.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def initialize_cluster_settings(client: OpenSearch) -> None:
         raise Exception("Failed to initialize cluster settings.")
 
 
-def init_index_template(client: MLClient, template_name = "nlp-template") -> None:
+def init_index_template(client: MLClient, template_name = "nlp-template", template_data=None) -> None:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:54]: Here we can drop template data as I think template name may be the only variable. We may pull out index patterns as well. I'll leave that decision to you.

File: demo/docbot/cluster_bootstrap.py [hunk 3]
```diff
@@ -60,53 +60,54 @@ def init_index_template(client: MLClient, template_name = "nlp-template") -> Non
...
-    }
 
-    response = client._client.indices.put_template(name=template_name, body=template)
+    response = client._client.indices.put_template(name=template_name, body=template_data)
     if not response["acknowledged"]:
       raise Exception("Unable to create index template.")
 
 
-def initialize_model_group(client: MLClient) -> None:
+def initialize_model_group(client: MLClient, model_group_name="Cohere_Group", data=None) -> None:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:92]: Here I think the group name should be the only variable in here.

File: demo/docbot/cluster_bootstrap.py [hunk 4]
```diff
@@ -60,53 +60,54 @@ def init_index_template(client: MLClient, template_name = "nlp-template") -> Non
...
     response = client.get_model_group_id(model_group_name=model_group_name)
 
     # Check if the Cohere model group already exists
     if response is not None:
-        data = {
-            "name": model_group_name,
-            "description": "Public Cohere Model Group",
-            "access_mode": "public"
-        }
+        if data is None:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:105]: What we can do to make data a little more modular is we can make the model group name appear in the description with f-strings so each would have the right name if we ever changed the model group name.

File: demo/docbot/cluster_bootstrap.py [hunk 5]
```diff
@@ -119,40 +120,41 @@ def initialize_model_group(client: MLClient) -> None:
 
 
 
-def initialize_connector(client: MLClient) -> None:
+def initialize_connector(client: MLClient, connector_name="Cohere Connector", connector_data=None) -> None:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:123]: Here we can remove the connector_data.

File: demo/docbot/cluster_bootstrap.py [hunk 6]
```diff
@@ -119,40 +120,41 @@ def initialize_model_group(client: MLClient) -> None:
...
-        },
-        "actions": [{
-            "action_type": "predict",
-            "method": "POST",
-            "url": "https://api.cohere.ai/v1/embed",
-            "headers": {
-                "Authorization": "Bearer ${credential.cohere_key}"
+    if connector_data is None:
+        connector_data = {
+            "name": "Cohere Connector",
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:133]: We should probably be using the variable name here. Then in the description below we can use f-strings again to get the connector name to show correctly there.

File: demo/docbot/cluster_bootstrap.py [hunk 7]
```diff
@@ -165,21 +167,22 @@ def initialize_connector(client: MLClient) -> None:
         print("Connector 'Cohere Connector' already exists. Skipping initialization.")
 
 
-def initialize_model(client: MLClient) -> None:
+def initialize_model(client: MLClient, model_data=None) -> None:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:170]: We can remove the model_data here and can instead just pull "name" out and have that be variable.

File: demo/docbot/cluster_bootstrap.py [hunk 8]
```diff
@@ -204,38 +207,39 @@ def initialize_model(client: MLClient) -> None:
 
 
 
-def initialize_ingestion_pipeline(client: MLClient):
+def initialize_ingestion_pipeline(client: MLClient, pipeline="cohere-ingest-pipeline", pipeline_data=None):
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:210]: Here we can remove pipeline data. Again below in the description we can make it more modular by including our variable pipeline name in the description.

File: demo/docbot/cluster_bootstrap.py [hunk 9]
```diff
@@ -244,48 +248,49 @@ def initialize_ingestion_pipeline(client: MLClient):
 
 
 
-def initialize_index(client: MLClient) -> None:
+def initialize_index(client: MLClient, index="cohere-index", index_data=None) -> None:
```
Comment (dtaivpp) [inline demo/docbot/cluster_bootstrap.py:251]: Here we can remove index_data and maybe we should lift the default pipeline as variable.

**Review by dtaivpp — CHANGES_REQUESTED** [review]
Sorry that the issue was too vague and that it took me till now to get feedback in. Let me know if you have any questions!

**Pull request closed without merging**